From e9c3cffe8ecb3951e7b5960eb99a25b8577f261b Mon Sep 17 00:00:00 2001
From: Chuanxiao Dong <chuanxiao.dong@intel.com>
Date: Wed, 30 Jul 2014 10:38:54 +0800
Subject: [PATCH 09/15] mmc: optimize mmc device power up ramp up time

Replace msleep() with usleep_range() to reduce
the mmc sleep time. Only for >20ms sleep, use
msleep.

Change-Id: I7f7693a58649a9749f94853171dd2ca4d54fc867
Signed-off-by: Chuanxiao Dong <chuanxiao.dong@intel.com>
---
 drivers/mmc/core/core.h | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/mmc/core/core.h b/drivers/mmc/core/core.h
index 443a584660f0..b7584d891fe3 100644
--- a/drivers/mmc/core/core.h
+++ b/drivers/mmc/core/core.h
@@ -55,9 +55,10 @@ static inline void mmc_delay(unsigned int ms)
 	if (ms < 1000 / HZ) {
 		cond_resched();
 		mdelay(ms);
-	} else {
+	} else if (ms < 20)
+		usleep_range(ms * 1000, (ms + 1) * 1000);
+	else
 		msleep(ms);
-	}
 }
 
 void mmc_rescan(struct work_struct *work);
-- 
1.9.1

